app/assets/javascripts/discourse/app/components/modal/bookmark.js

```diff
@@ -0,0 +1,352 @@
+import Component from "@glimmer/component";
+import { extractError } from "discourse/lib/ajax-error";
+import { sanitize } from "discourse/lib/text";
+import { CLOSE_INITIATED_BY_CLICK_OUTSIDE } from "discourse/components/d-modal";
+import { inject as service } from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+import { now, parseCustomDatetime, startOfDay } from "discourse/lib/time-utils";
+import { AUTO_DELETE_PREFERENCES } from "discourse/models/bookmark";
+import I18n from "I18n";
+import KeyboardShortcuts from "discourse/lib/keyboard-shortcuts";
+import ItsATrap from "@discourse/itsatrap";
+import { Promise } from "rsvp";
+import {
+  TIME_SHORTCUT_TYPES,
+  defaultTimeShortcuts,
+} from "discourse/lib/time-shortcut";
+import { action } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { formattedReminderTime } from "discourse/lib/bookmark";
+import { and, notEmpty } from "@ember/object/computed";
+import discourseLater from "discourse-common/lib/later";
+
+const BOOKMARK_BINDINGS = {
+  enter: { handler: "saveAndClose" },
+  "d d": { handler: "delete" },
+};
+
+export default class BookmarkModal extends Component {
+  @service dialog;
+  @service currentUser;
+  @service site;
+
+  @tracked postDetectedLocalDate = null;
+  @tracked postDetectedLocalTime = null;
+  @tracked postDetectedLocalTimezone = null;
+  @tracked prefilledDatetime = null;
+  @tracked flash = null;
+  @tracked flashType = "error";
```

Comment:
We can drop this here in favor of adding the string directly to the `<DModal>`

```
<DModal @flashType="error">
``` 

as it doesn't look like we are manipulating the value.